aws-api/src/main/java/com/amplifyframework/api/aws/SubscriptionEndpoint.java

```diff
@@ -317,27 +325,35 @@ private String buildConnectionRequestUrl() throws ApiException {
             this.responseType = responseType;
             this.subscriptionReadyAcknowledgment = new CountDownLatch(1);
             this.subscriptionCompletionAcknowledgement = new CountDownLatch(1);
+            this.failed = false;
         }
 
         void acknowledgeSubscriptionReady() {
             subscriptionReadyAcknowledgment.countDown();
         }
 
+        void acknowledgeSubscriptionFailure() {
+            failed = true;
+            subscriptionReadyAcknowledgment.countDown();
+        }
+
         boolean awaitSubscriptionReady() {
             try {
                 if (!subscriptionReadyAcknowledgment.await(ACKNOWLEDGEMENT_TIMEOUT, TimeUnit.SECONDS)) {
                     dispatchError(new ApiException(
-                        "Subscription not acknowledged.",
+                        "Timed out waiting for subscription start_ack.",
                         AmplifyException.TODO_RECOVERY_SUGGESTION
                     ));
                     return false;
+                } else if (failed) {
+                    // An error was already dispatched at the time of failure, so don't dispatch a second one.
+                    return false;
                 }
             } catch (InterruptedException interruptedException) {
-                dispatchError(new ApiException(
-                    "Failure awaiting subscription acknowledgement.",
-                    interruptedException,
-                    AmplifyException.TODO_RECOVERY_SUGGESTION
-                ));
+                // Triggered when the Future created in SubscriptionOperation is cancelled, which happens when the
+                // subscription Observable is disposed, which happens when a SUBSCRIPTION_ERROR occurs.  Don't dispatch
+                // any error because the caller has likely already been disposed.
+                LOG.warn("Thread interrupted awaiting subscription acknowledgement.", interruptedException);
```

Comment:
For the scenario where I observed this error happening, no.  The `InterruptedException` is thrown because `cancel` was called on the `SubscriptionOperation`, which then tries to [cancel the Future](https://github.com/aws-amplify/amplify-android/blob/0d560a667f0a856fbe72b543622dd3166d11bab2/aws-api/src/main/java/com/amplifyframework/api/aws/SubscriptionOperation.java#L111).

There may be other scenarios that could cause an `InterruptedException`, but I think they should probably be handled similarly - stop, do any cleanup that you can, and return immediately.  Dispatching an error to the observer fails because the observer has already been disposed. 

Note though that this particular change is not needed for "swallowing unauthorized errors for subscriptions" since the errors are swallowed, and an interruption is prevented.   This change just improves handling of other errors that are not swallowed.